pygmt/src/grdview.py

```diff
@@ -154,24 +152,19 @@ def grdview(self, grid, **kwargs):
     >>> fig.show()
     """
     kwargs = self._preprocess(**kwargs)  # pylint: disable=protected-access
-    with Session() as lib:
-        file_context = lib.virtualfile_from_data(check_kind="raster", data=grid)
 
-        with contextlib.ExitStack() as stack:
-            if kwargs.get("G") is not None:
-                # deal with kwargs["G"] if drapegrid is xr.DataArray
-                drapegrid = kwargs["G"]
-                if data_kind(drapegrid) in ("file", "grid"):
-                    if data_kind(drapegrid) == "grid":
-                        drape_context = lib.virtualfile_from_data(
-                            check_kind="raster", data=drapegrid
-                        )
-                        kwargs["G"] = stack.enter_context(drape_context)
-                else:
-                    raise GMTInvalidInput(
-                        f"Unrecognized data type for drapegrid: {type(drapegrid)}"
-                    )
-            fname = stack.enter_context(file_context)
+    if kwargs.get("G") is not None and data_kind(kwargs["G"]) not in ("file", "grid"):
+        raise GMTInvalidInput(
+            f"Unrecognized data type for drapegrid: {type(kwargs['G'])}"
+        )
+
+    with Session() as lib:
+        with lib.virtualfile_from_data(
+            check_kind="raster", data=grid
+        ) as fname, lib.virtualfile_from_data(
+            check_kind="raster", data=kwargs.get("G"), optional_data=True
+        ) as drapegrid:
+            kwargs["G"] = drapegrid
```

Comment:
Slightly relevant, how would we handle RGB xarray.DataArray inputs to drapegrid `-G` as mentioned at #1467? Currently [GMT 6.4.0](https://docs.generic-mapping-tools.org/6.4/grdview.html#g) requires passing it in as `-Ggrd_r -Ggrd_g -Ggrd_b`, but I'm wondering if we should only support just a single `xarray.DataArray` of shape (3, Y, X) instead.

Reply:
> we should only support just a single `xarray.DataArray` of shape (3, Y, X) instead.

Sounds good to me.